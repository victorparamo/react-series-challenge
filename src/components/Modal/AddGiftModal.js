import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import giftModalActions from '../../redux/actions/giftModal';
import favoriteActions from '../../redux/actions/favorites';
import './Modal.css';

Modal.setAppElement('#root');

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    width: '210px',
    height: '310px',
    overflow: 'hidden',

    transform: 'translate(-50%, -50%)'
  }
};

const checkFavorite = (props) =>{
    const {
        gift,
        favorites
    } = props;

    return Object.keys(favorites).includes(gift.name);
}

const AddGiftModal = (props) => {
  const {
    isModalOpen,
    gift,
    hideGiftModal,
    addFavorite,
    deleteFavorite,
    favorites,
  } = props;

  const {
      url, 
      name
  } = gift 

  const isFavorite = checkFavorite({favorites,gift});

  return (
    <Modal
      isOpen={isModalOpen}
      style={customStyles}
    >
      <div>
        <header className="modal-header">{name}</header>
        <img className="modal-image" src={url} alt={""}/>
      </div>
      <button className="modal-button" onClick={() => hideGiftModal()}>Close</button>
      <button className={`modal-button ${isFavorite ? "favorite":""}`} onClick={isFavorite ? ()=>deleteFavorite(gift): ()=>addFavorite(gift)}>Like</button>
      <span>{}</span>
    </Modal>
  );
}

const mapStateToProps = (state) => {
  const { isModalOpen, gift } = state.isModalOpen;
  const { favorites, } = state.favoritesReducer;

  return {
    isModalOpen,
    gift,
    favorites,
  };
};

const mapDispatchToProps = (dispatch) => {
  const { hideGiftModal } = giftModalActions.creators;
  const { addFavorite, deleteFavorite  } = favoriteActions.creators;

  return bindActionCreators({
    hideGiftModal,
    addFavorite,
    deleteFavorite,
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(AddGiftModal);
