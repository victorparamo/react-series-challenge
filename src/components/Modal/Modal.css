.modal-button {
  width: 80px;
  height: 20px;
  border: 1px solid #000;
  font-family: inherit;
  font-size: 10px;
  margin-top: 10px;
  margin-left: 20px;
  background: none;
  text-transform: uppercase;
}
.favorite {
    background-color:#1E90FF;
}

.modal-image {
    width: 180px;
    border-radius: 10px;
    margin-left: 15px;
}
.modal-button:hover {
    cursor: pointer;
}

.modal-header {
    text-align: center;
    font-family: inherit;
    text-transform: uppercase;
    width: 70%;
    margin: auto;
    margin-bottom: 10px;
}