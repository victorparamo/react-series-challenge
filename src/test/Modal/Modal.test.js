import React from 'react';
import { shallow } from 'enzyme';
import { AddGiftModal } from '../../components/Modal/AddGiftModal';

describe('test', () => {
  it('will work', () => {
    const component = shallow(<AddGiftModal isModalOpen={true} gift={{"url":"cdcdxs", "name":"dcdjc"}} favorites={{"cdcd":"dcdcd"}}/>);
    expect(component).toMatchSnapshot();
  });
});
