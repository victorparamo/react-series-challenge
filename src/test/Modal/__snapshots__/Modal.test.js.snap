// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test will work 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favorites={
        Object {
            "cdcd": "dcdcd",
          }
    }
    gift={
        Object {
            "name": "dcdjc",
            "url": "cdcdxs",
          }
    }
    isModalOpen={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        <styled.img
          alt=""
          src="cdcdxs"
/>,
        <styled.div>
          <styled.p>
                    dcdjc
          </styled.p>
          <styled.span
                    onClick={[Function]}
          >
                    Close
          </styled.span>
          <styled.span
                    className=""
                    onClick={[Function]}
          >
                    Like
          </styled.span>
</styled.div>,
      ],
      "closeTimeoutMS": 0,
      "isOpen": true,
      "parentSelector": [Function],
      "portalClassName": "ReactModalPortal",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
      "style": Object {
        "content": Object {
          "bottom": "auto",
          "height": "auto",
          "left": "50%",
          "overflow": "auto",
          "padding": "0",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "210px",
        },
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "",
          "src": "cdcdxs",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.p>
              dcdjc
</styled.p>,
            <styled.span
              onClick={[Function]}
>
              Close
</styled.span>,
            <styled.span
              className=""
              onClick={[Function]}
>
              Like
</styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "dcdjc",
            },
            "ref": null,
            "rendered": "dcdjc",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Like",
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Like",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          <styled.img
            alt=""
            src="cdcdxs"
/>,
          <styled.div>
            <styled.p>
                        dcdjc
            </styled.p>
            <styled.span
                        onClick={[Function]}
            >
                        Close
            </styled.span>
            <styled.span
                        className=""
                        onClick={[Function]}
            >
                        Like
            </styled.span>
</styled.div>,
        ],
        "closeTimeoutMS": 0,
        "isOpen": true,
        "parentSelector": [Function],
        "portalClassName": "ReactModalPortal",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
        "style": Object {
          "content": Object {
            "bottom": "auto",
            "height": "auto",
            "left": "50%",
            "overflow": "auto",
            "padding": "0",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "210px",
          },
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "src": "cdcdxs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.p>
                dcdjc
</styled.p>,
              <styled.span
                onClick={[Function]}
>
                Close
</styled.span>,
              <styled.span
                className=""
                onClick={[Function]}
>
                Like
</styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "dcdjc",
              },
              "ref": null,
              "rendered": "dcdjc",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Close",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Like",
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Like",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
